Exiled.Events/EventArgs/ChangingRoleEventArgs.cs

```diff
@@ -75,5 +78,53 @@ public ChangingRoleEventArgs(Player player, RoleType newRole, bool shouldPreserv
         /// Gets or sets a value indicating whether the event can continue.
         /// </summary>
         public bool IsAllowed { get; set; } = true;
+
+        /// <summary>
+        /// Conslidates all the player information to be easily access through helper functions later on.
+        /// </summary>
+        private void ConsolidatePlayerInformation()
+        {
+            if (Player.InstantiatedRolesToPlayers.TryGetValue(Player.Role, out HashSet<Player> previousRolePlayers))
+            {
+                previousRolePlayers.Remove(Player);
+            }
+
+            if (Player.InstantiatedRolesToPlayers.TryGetValue(NewRole, out HashSet<Player> newRolePlayers))
+            {
+                newRolePlayers.Add(Player);
+            }
+            else
+            {
+                Player.InstantiatedRolesToPlayers.Add(NewRole, new HashSet<Player>() { Player });
```

Comment:
Create all this value when server start with empty  hashset and so you will not need to make this method like an yandere dev :3

Reply:
This behavior prevents developer mistakes. I already considered allocating all roles, side, team preemptively; however, if a developer messes it up or someone misses a new role, side, or team - this code will not run into a problem as it dynamic. I would make this change if it is a consensus choice

Reply:
Developer should don't have access to this things just make it private

Reply:
Hmm fair point, I agree then - I'll make change tomorrow,  Thanks!